### PR TITLE
Add explicit defaults for RELEASE_TYPE and RELEASE_SCOPE

### PR DESCRIPTION
**Pull request by Joaquimmnetto**
## Description

Adds explicit default values for `RELEASE_TYPE` and `RELEASE_SCOPE` parameters, because jenkins doesn't set any parameters on first pipeline run, and this may lead to issues in the publish step.

This was done by setting new envs RELEASE_TYPE and RELEASE_SCOPE in the setup step, because the `params` map provided by jenkins was immutable. 

There are a bunch of changes, but most of them are repetitive so, don't be intimidated :)
 

## Changes
* ![FIX] Pipelines breaking on publish if in first jenkins run.

[NEW]:https://resources.atlas.wooga.com/icons/icon_new.svg "New"
[ADD]:https://resources.atlas.wooga.com/icons/icon_add.svg "Add"
[IMPROVE]:https://resources.atlas.wooga.com/icons/icon_improve.svg "IMPROVE"
[CHANGE]:https://resources.atlas.wooga.com/icons/icon_change.svg "Change"
[FIX]:https://resources.atlas.wooga.com/icons/icon_fix.svg "Fix"
[UPDATE]:https://resources.atlas.wooga.com/icons/icon_update.svg "Update"

[BREAK]:https://resources.atlas.wooga.com/icons/icon_break.svg "Remove"
[REMOVE]:https://resources.atlas.wooga.com/icons/icon_remove.svg "Remove"
[IOS]:https://resources.atlas.wooga.com/icons/icon_iOS.svg "iOS"
[ANDROID]:https://resources.atlas.wooga.com/icons/icon_android.svg "Android"
[WEBGL]:https://resources.atlas.wooga.com/icons/icon_webGL.svg "WebGL"
[UNITY]:https://resources.atlas.wooga.com/icons/icon_unity.svg "Unity"
[GRADLE]:https://resources.atlas.wooga.com/icons/icon_gradle.svg "Gradle"
[LINUX]:https://resources.atlas.wooga.com/icons/icon_linux.svg "Linux"

### CODE REVIEW

**Comment by Azurelol** [comment]
Seems alright to me, though I wonder whether mapping params to env is fine. I would like to hear what Manfred has to say about it.

File: vars/javaLibs.groovy [hunk 1]
```diff
@@ -39,6 +39,10 @@ def call(Map configMap = [:], Closure stepsConfigCls) {
       stage('Preparation') {
         agent any
         steps {
+          script {
+            env.RELEASE_TYPE = params.RELEASE_TYPE?: "snapshot"
+            env.RELEASE_SCOPE = params.RELEASE_SCOPE?: ""
```
Comment (Larusso) [inline vars/javaLibs.groovy:44]: I would wish for a more formulized way of setting the default. Could we at least define a const value per file with the defaults so we use this in the params block and here?

**Review by Larusso — COMMENTED** [review]
It's quite the change. As far as I understood it it failed only in the new pipelines or? So why bother going over all other pipelines which did not have this issue. I mean from a don't break what is not broken kind of a standpoint here.

File: vars/javaLibs.groovy (hunk 1 above)
Comment (Joaquimmnetto) [reply vars/javaLibs.groovy]: Done.

**Review by Larusso — APPROVED** [review]

**Pull request merged**